provider/datagen/src/transform/cldr/source.rs

```diff
@@ -9,71 +9,55 @@ use std::fmt::Debug;
 use std::str::FromStr;
 
 /// Specifies a variant of CLDR JSON
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, serde::Deserialize)]
 #[non_exhaustive]
 pub enum LocaleSubset {
-    /// Includes all data
+    /// Deprecated. Use `Modern` + `Moderate` + `Basic`
+    #[deprecated]
+    #[serde(skip)]
```

Comment:
I thought we had discussed making "full" mean "enumerate the directory", and the other three as the ones that look inside coverageLevels.json

Reply:
This is from the discussion:

* `icu4x-datagen --locales full`
  * Include all locales in the zip/directory (i.e. "basic", "moderate", and "modern")
* `icu4x-datagen --locales modern`
  * Include "modern" locales according to coverageLevels.json
* `icu4x-datagen --locales moderate`
  * Include "moderate" locales according to coverageLevels.json
* `icu4x-datagen --locales basic`
  * Include "basic" locales according to coverageLevels.json
* `icu4x-datagen --locales core`
  * Include locales without level in coverageLevels.json; locales that don’t even fulfil “basic”
* `icu4x-datagen –locales modern moderate`
  * Include "modern" and "moderate" locales according to coverageLevels.json

On the command line `--locales full` is available. However, I think in the API it's clearer to use `source.locales(&[::Basic, ::Moderate, ::Modern])` than `source.locales(&[::Full])`. The enum variants are disjoint, and `Full` violates that.

Reply:
> Include all locales in the zip/directory (i.e. "basic", "moderate", and "modern")

I interpret the non-parenthetical part "all locales in the zip/directory" with more weight than the parenthetical i.e.

How are you supposed to use the API if you want the directory enumeration behavior?

Reply:
I was working under the assumption that all locales are listed in `coverageLevels.json`.

Enumeration behaviour is still available through passing `locales = None` to the API. We can make `full` do that on the CLI as well, but then it wouldn't necessarily be the same as basic + modern + moderate. In any case, we don't need `LocaleSubset::Full` for this.